x-pack/plugins/infra/public/test_utils/use_global_storybook_theme.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { StoryContext } from '@storybook/addons';
+import React, { useEffect, useMemo, useState } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import type { CoreTheme } from '../../../../../src/core/public';
+import { EuiThemeProvider } from '../../../../../src/plugins/kibana_react/common';
+import { KibanaThemeProvider } from '../../../../../src/plugins/kibana_react/public';
+
+export const useGlobalStorybookTheme = ({ globals: { euiTheme } }: StoryContext) => {
+  const theme = useMemo(() => euiThemeFromId(euiTheme), [euiTheme]);
+  const [theme$] = useState(() => new BehaviorSubject(theme));
+
+  useEffect(() => {
+    theme$.next(theme);
+  }, [theme$, theme]);
+
+  return {
+    theme,
+    theme$,
+  };
+};
+
+export const GlobalStorybookThemeProviders: React.FC<{ context: StoryContext }> = ({
+  context,
```

Comment:
Nit: Would it be better to simply name this variable storyContext to make it clear it's not a React `Context`?